vsintegration/src/FSharp.Editor/Hints/InlineParameterNameHints.fs

```diff
@@ -29,7 +39,12 @@ module InlineParameterNameHints =
         symbol.IsFunction
         && isNotBuiltInOperator // arguably, hints for those would be rather useless
 
-    let getHints 
+    let isUnionCaseValidForHint (symbol: FSharpUnionCase) (symbolUse: FSharpSymbolUse) =
+        // is the union case being used as a constructor and is it not Cons
```

Comment:
So this seems to be sufficient to avoid any further explosions, but maybe @T-Gro @vzarytovskii you have any extra thoughts on some similar situations.